app/v1/models/response/listPayments/ListPaymentsResponse.scala

```diff
@@ -16,13 +16,35 @@
 
 package v1.models.response.listPayments
 
+import cats.Functor
+import config.AppConfig
 import play.api.libs.json._
+import v1.hateoas.{HateoasLinks, HateoasListLinksFactory}
+import v1.models.hateoas.{HateoasData, Link}
 
-case class ListPaymentsResponse(payments: Seq[Payment])
+case class ListPaymentsResponse[I](payments: Seq[I])
 
-object ListPaymentsResponse {
-  implicit val reads: Reads[ListPaymentsResponse] =
-    (JsPath \ "paymentDetails").read[Seq[Payment]].map(ListPaymentsResponse(_))
+object ListPaymentsResponse extends HateoasLinks {
 
-  implicit val writes: OWrites[ListPaymentsResponse] = Json.writes[ListPaymentsResponse]
+  implicit def reads[I: Reads]: Reads[ListPaymentsResponse[I]] = (JsPath \ "paymentDetails").read[Seq[I]].map(ListPaymentsResponse(_))
```

Comment:
Do we need an 'implicitly' here to make sure the compiler can find the Payments reads? I know that in the past we have done something like 
```suggestion
  implicit def reads[I: Reads]: Reads[ListPaymentsResponse[I]] = 
  implicitly[Reads[Seq[I]]].map(ListPaymentsResponse(_))
```
but I haven't seen it done with an actual JsPath before.

Reply:
Not required in this scenario, but nothing wrong adding it.